validator/rules/executor.go

```diff
@@ -113,28 +116,22 @@ func (config *RuleConfig) initDefaultRules() {
 		},
 	}
 
-	// Check verbose input from user
-	var verbose bool
-	if config != nil && config.Verbose {
-		verbose = true
+	if err := mergo.Merge(&config.Length, defaultConfig.Length); err != nil {
```

Comment:
I think overal config is confusing. WE should either:

- Add rules field with array of IRule
- Put string Rule after each rule(not like it personally)

Reply:
okay, so I really need to rethink the architecture, as everything is dependent on config. I will refactor it, keeping the functionality same!

I have one more concern - here we don't really need validator package, shall we do everything under rules package? But this can be changed while doing above refactoring!

Reply:
Yes.. No pressure. Take it easy

Reply:
If the aim here is to merge two config files into one giving precedence to a passed in file but keeping default values were no values are passed in and we are already using `mergo`. We could try using merge with the transformer: `mergo.Merge(&config.Length, defaultConfig.Length, mergo.WithOverride)`. This should keep config with first values whilst only updating fields passed in from `defaultConfig.Length`.

Speaking of `.Length` is this the best name for this field? is it a validated ( element size wise ) map of config fields? I may just not be getting it either, in which case my suggestion may not be applicable.

Reply:
I got you point @dimakis. But we want to keep defaultConfig.Length as it is and merge config.Length into it. We can do opposite as you suggested but then we need to do `*config = *defaultConfig`, extra lines of code. This can be avoided I'd say.

> Speaking of .Length is this the best name for this field? is it a validated ( element size wise ) map of config fields? I may just not be getting it either, in which case my suggestion may not be applicable.

Length is a struct of Limits (that is map[string]Limit) and Limit is struct consisting of Max, Min length of field(int value). We can think over the names again.

> Why we would merge specific rules. Can we merge all config paths?

Problem was mergo cannot merge 2 slices. It can only merge structs and maps. And the previous method also had similar problems.

Reply:
> I got you point @dimakis. But we want to keep defaultConfig.Length as it is and merge config.Length into it. We can do opposite as you suggested but then we need to do `*config = *defaultConfig`, extra lines of code. This can be avoided I'd say. And btw this code might change as there is some major refactoring going on.
> It may be one extra assignment, but it negates the need to check for duplication of values in a slice -> `removeDuplicates()` becomes obsolete.
> > Speaking of .Length is this the best name for this field? is it a validated ( element size wise ) map of config fields? I may just not be getting it either, in which case my suggestion may not be applicable.
> 
> Length is a struct of Limits (that is map[string]Limit) and Limit is struct consisting of Max, Min length of field(int value). We can think over the names again.
>  +1